spec/registry.md

```diff
@@ -422,6 +422,12 @@ The following options and their values are required to be available on the funct
 - `maximumSignificantDigits`
   - ([digit size option](#digit-size-options))
 
+If the _operand_ of the _expression_ is an implementation-defined type,
+such as the _resolved value_ of an _expression_ with a `:number` or `:integer` _annotation_,
```

Comment:
We are defining `:number` and `:integer` here. Is "such as" needed?

Reply:
The "such as" is making sure that the _resolved value_ type is included in the "an implementation-defined type" set of possibilities.

Reply:
Presumably it would be that, otherwise the result would be bad operand?

---

Why does the operand need to be of the implementation-defined variety? Aren't option values also applicable to, for example, accepted literal values?

```
{-1.234 :number maximumFractionDigits=2 signDisplay=always}
```

Why wouldn't the options "stick to" the value?

Reply:
I'm trying to follow our practice of not defining exactly what a "number" type is, while allowing for some implementation-defined number types, such as the resolved value coming from a `:number` annotation, to contain options within the value of the operand.

So this is covering each of the following use cases:

1. `{$x :number}`
where in a JS context that could be formatted with `{ x: BigInt(42) }`, i.e. using a language-specific numerical type.

2. `{$y :number style=currency}`
where in a JS context that could be formatted with
`{ y: Object.assign(new Number(42), { options: { currency: 'EUR' } }) }`,
i.e. an object that carries both a numerical value and a bag of options.

3. `.local $z = {42 :number notation=compact} {{{$z :number compactDisplay=long}}}`
so that in the _placeholder_ the options of the _operand_ `$z` can be combined with the _expression_ _options_.

Reply:
Right. We're on the same page technically. I'm bikeshedding the language.

I think the challenge for me is that "implementation-defined type" appears to exclude some values (such as literal representation). It's not clear to me if we require the implementation to resolve it to an implementation-defined type. For example, we say:

> The operand of a number function is either an implementation-defined type or a literal whose contents match the number-literal production in the [ABNF](https://github.com/unicode-org/message-format-wg/blob/main/spec/message.abnf). All other values produce a Bad Operand error.

... but we don't say that the literal _becomes_ one of the implementation-defined types (although it might).

Reply:
The thing is, though, that a literal representation of a number cannot include option values within the operand. The _annotation_ of a literally represented number can absolutely have options, but its _operand_ cannot.

Reply:
What about transitivity?

```
.local $a = {42 :number signDisplay=always}
.local $b = {$a :number minimumFractionDigits=1}
{{{$b}'s operand is an annotated literal??}}
```

Reply:
In that example, when resolving the _expression_ defining the value of `$b`, its _operand_ is a _resolved value_, an implementation-defined type. The numerical value contained in that implementation-defined type is the number 42, and it includes a bag of resolved options `{ signDisplay: 'always' }`.

When resolving the value of `$b` or formatting the placeholder `{$b}`, the only impact that the literal origin of the value should have is that the fallback string that's used for it is `|42|`, and not e.g. `$x`.